docs/source/devguide/tests.rst

```diff
@@ -59,6 +59,26 @@ If you want to view testing output on failure run::
 
     ctest --output-on-failure
 
+Possible Reasons for Test Failures
+----------------------------------
+
+You may find that when you run the test suite, not everything passes. First,
+make sure you have satisfied all the prerequisites above. After you have done
+that, consider the following:
+
+- When building OpenMC, make sure you run CMake with
+  ``-DCMAKE_BUILD_TYPE=Debug``. Building with a release build will result in
+  some test failures due to differences in which compiler optimizations are
+  used.
+- If you are running on a CPU with many cores, you may need to limit the number
+  of OpenMP threads used. You can set the :envvar:`OMP_NUM_THREADS` environment
+  variable to 2.
```

Comment:
Can you add a quick explanation here about why it's necessary to limit to 2 threads when running the tests? (non-associative floating point ops)

Reply:
Done :heavy_check_mark: